solidity/contracts/deposit/DepositUtils.sol

```diff
@@ -489,33 +523,32 @@ library DepositUtils {
         bool frtExists = feeRebateTokenHolder(_d) != address(0);
         bool redeemerHoldsTdt = depositOwner(_d) == _redeemer;
 
-        return computeSignerFee(_d, preTerm, redeemerHoldsFrt, !redeemerHoldsTdt, frtExists);
+        return computeSignerFee(_d, preTerm, redeemerHoldsFrt, redeemerHoldsTdt, frtExists);
     }
 
     /// @notice  Get fees owed for redemption
     /// @param _preTerm              True if the Deposit is not at-term and not in courtesy_call.
     /// @param _redeemerHoldsFrt     True if the redeemer holds the FRT.
-    /// @param _redeemerIsThirdParty True if the redeemer is not the TDT owner.
+    /// @param _redeemerHoldsTdt     True if the redeemer is the TDT owner.
     /// @param _frtExists            True if the FRT exists.
     /// @return                      The fees owed in TBTC.
     function computeSignerFee(
         DepositUtils.Deposit storage _d,
         bool _preTerm,
         bool _redeemerHoldsFrt,
-        bool _redeemerIsThirdParty,
+        bool _redeemerHoldsTdt,
         bool _frtExists
     ) internal view returns (uint256) {
-        bool redeemerOwesNoFee;
+        bool redeemerOwesNoFeeRebate;
         if (_preTerm) {
-            redeemerOwesNoFee = _redeemerHoldsFrt;
-        } else {
-            redeemerOwesNoFee = _frtExists || _redeemerIsThirdParty;
+            redeemerOwesNoFeeRebate = _redeemerHoldsFrt;
+        }else {
+            redeemerOwesNoFeeRebate = true;
```

Comment:
You can drop this branch by setting `redeemerOwesNoFeeRebate = true` at initialization. Then we replace it with `_redeemerHoldsFrt` when `_preTerm` is true.

In fact, assuming this logic is correct, I think this can now be `bool redeemerOwesNoFeeRebate = atTerm || _redeemerHoldsFrt;` and the if statement can go away altogether.

Reply:
Second part of this that doesn't hit right is that the redeemer should never really owe the fee rebate, it's just that the escrow requires it. I'm wondering if `computeSignerFee` becomes `computeRedemptionFeeEscrow`. Then:

```solidity
function computeRedemptionFeeEscrow(
    DepositUtils.Deposit storage _d,
    bool _preTerm,
    bool _redeemerHoldsFrt
) internal view returns (uint256) {
    bool escrowRequiresFeeRebate = _preTerm && !_redeemerHoldsFrt;

    if (escrowRequiresFeeRebate) {
        return signerFee(_d).mul(2);
    } else {
        return signerFee(_d);
    }
}
```

Reply:
the simplification makes sense, and the rename also makes sense I think